src/index.ts

```diff
@@ -84,10 +83,10 @@ interface ITargetSourceDepth {
  */
 export function deepMerge(target: Object, source: Object, depth = 10): Object {
   // Keep track of seen objects to detect recursive references.
-  const seen: Array<Object> = [];
+  const seen: Object[] = [];
```

Comment:
I have started to collect some research on the topic at https://github.com/openinf/util-object/wiki/Type-Annotation-for-JavaScript-Objects, which we'll be incorporating as part of our style guide. Once I have settled on our preference, i'll be sure to make the update here.

/cc @septs as this would be another guideline to be added to our JS authoring guide (and enforced via our standard) ^^

Reply:
`Object` -> `object`
`Number` -> `number`
`String` -> `string`
`BigInt` -> `bigint`
`Symbol` -> `symbol`
`Boolean` -> `boolean`

Reply:
> `Object` -> `object`

For this one, that would seem alright, but it would be very loose as it would simply mean any non-nullish value. Wonder what you think about the [TypeScript `Record` utility type](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type)?

Reply:
I don't like that it is called the same as [`Record` from the TC39 proposal](https://tc39.es/proposal-record-tuple/tutorial/), seeing as how they have different semantics iiuc. The TypeScript `Record` utility type isn't an immutable object afaik, so mixing that in our codebase is a bit annoying imo.